src/runtime/msg/internal.rs

```diff
@@ -43,12 +44,21 @@ pub enum Internal {
     Logout,
     /// Dispatched when addons needs to be installed.
     InstallAddon(Descriptor),
+    /// Dispatched when a new stream is loaded into the Player.
+    StreamLoaded {
+        stream: Stream,
+        meta_id: Option<String>,
+        video_id: Option<String>,
+        transport_url: Option<Url>,
+    },
```

Comment:
What are the use-cases for when we have `StreamLoaded` but no meta_id or video_id or transport_url?